libres/lib/analysis/analysis_module.cpp

```diff
@@ -36,12 +36,13 @@
 
 #include <fmt/format.h>
 #include <ert/logging.hpp>
+#include <ert/python.hpp>
 
 auto logger = ert::get_logger("analysis");
 
 struct analysis_module_struct {
-    std::unique_ptr<ies::data::Data> module_data;
-    std::unique_ptr<ies::config::Config> module_config;
+    std::shared_ptr<ies::data::Data> module_data;
```

Comment:
There is a clear owner here - I would recommend staying with the `unique_ptr` and use `py::return_value_policy::reference_internal` when returning to Python.

Reply:
@pinkwah Have a compelling argument, as to why we are using shared ptr. This avoids having dangling pointers in python in a case where the analysis module would be destroyed in C++

Reply:
Well - that is the difference between `return_value_policy::reference` and `return_value_policy::reference_internal`- the latter adds a "keep alive" policy on the parent. That said this machinery might be fooled by our cwrap based not-really-standard calling convention. Anyway - I agree `shared_ptr` is a safe alteranative.

Reply:
A benefit of using a `shared_ptr` is that you don't need to keep the entire object alive when Python only cares about a subset of it.

But yes, `reference_internal` is likely to do the job. Although, in that case the preferred solution is to not use any smart pointers at all and just use object semantics within the struct.